src/foam/u2/filter/ReferenceFilterView.js

```diff
@@ -0,0 +1,266 @@
+          });
+        }
+
+        // Filter out selectedOptions
+        selectedOptions.forEach(function(selection) {
+          options = options.filter(function(option) {
+            return option !== selection;
+          });
+        });
+
+        return options;
+      }
+    },
+    {
+      class: 'String',
+      name: 'search'
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'dao',
+      required: true
+    },
+    {
+      name: 'daoContents',
+    },
+    {
+      name: 'idToStringDisplayMap',
+      documentation: 'map that contains ids as keys and names as values',
+      expression: function(referenceObjectsArray) {
+        if ( ! referenceObjectsArray ) return;
+        referenceObjectsArray = referenceObjectsArray.instance_.array;
+        var result = {};
+        for ( i =0; i < referenceObjectsArray.length; i++ ) {
+            if ( this.setOfReferenceIds.has(referenceObjectsArray[i].instance_.id) ) {
+              result[referenceObjectsArray[i].instance_.id] = referenceObjectsArray[i].legalName;
+            }
+          }
+        return result;
+      }
+    },
+    {
+      name: 'setOfReferenceIds',
+      documentation: 'an array of unique reference ids',
+      expression: function(daoContents) {
+        var result = [];
+        daoContents.instance_.array.forEach( function(id) {
+          ( result.push(id.instance_.owner) );
+        });
+        return new Set(result);
+      }
+    },
+    {
+      name: 'referenceObjectsArray',
+    },
+    {
+      name: 'name',
+      documentation: `Required by SearchManager.`,
+      value: 'Reference filter view'
+    },
+   {
+      name: 'predicate',
+      documentation: ``,
+      expression: function(selectedOptions, idToStringDisplayMap) {
+        function getKeyByValue_(object, value) {
+          return Object.keys(object).find( (key) => object[key] === value );
+        }
+
+        if ( selectedOptions.length <= 0 ) {
+          return this.True.create();
+        }
+
+        if ( selectedOptions.length === 1 ) {
+          return foam.mlang.predicate.Eq.create({
+            arg1: this.property,
+            arg2: parseInt(getKeyByValue_(idToStringDisplayMap, selectedOptions[0]))
+          });
+        }
+
+        var pred = foam.mlang.predicate.Or.create({ args: [] });
+        selectedOptions.forEach( (string) => {
+          pred.args.push(foam.mlang.predicate.Eq.create({
+            arg1: this.property,
+            arg2: parseInt(getKeyByValue_(idToStringDisplayMap, string))
+          }));
+        });
+        return pred;
+      }
+  }
+  ],
+
+
+  messages: [
+    { name: 'LABEL_PLACEHOLDER', message: 'Search' },
+    { name: 'LABEL_SELECTED', message: 'SELECTED OPTIONS' },
+    { name: 'LABEL_FILTERED', message: 'OPTIONS' }
+  ],
+
+
+    methods: [
+      async function initE() {
+        var self  = this;
+        this.daoContents = await this.dao.select(); //  gets contents from the source dao
```

Comment:
This should not be in an await. Because this will block the UI from rendering if the internet connection is slow.